go/enclave/enclave.go

```diff
@@ -217,18 +217,21 @@ func (e *enclaveImpl) Start(block types.Block) error {
 	return nil
 }
 
-func (e *enclaveImpl) ProduceGenesis(blkHash gethcommon.Hash) (common.BlockSubmissionResponse, error) {
+func (e *enclaveImpl) ProduceGenesis(blkHash gethcommon.Hash) (*common.BlockSubmissionResponse, error) {
 	rolGenesis, b := e.chain.ProduceGenesis(blkHash)
-	return common.BlockSubmissionResponse{
+	return &common.BlockSubmissionResponse{
 		ProducedRollup: rolGenesis.ToExtRollup(e.transactionBlobCrypto),
 		BlockHeader:    b.Header(),
-		IngestedBlock:  true,
 	}, nil
 }
 
 // SubmitBlock is used to update the enclave with an additional L1 block.
-func (e *enclaveImpl) SubmitBlock(block types.Block, isLatest bool) (common.BlockSubmissionResponse, error) {
-	bsr := e.chain.SubmitBlock(block, isLatest)
+func (e *enclaveImpl) SubmitBlock(block types.Block, isLatest bool) (*common.BlockSubmissionResponse, error) {
+	bsr, err := e.chain.SubmitBlock(block, isLatest)
+	e.logger.Trace("Processed block submission", "success", err == nil, "blk", block.Number(), "blkHash", block.Hash())
```

Comment:
Shouldn't you check the error before declaring this a success?

Reply:
It prints `success=true` or `success=false` based on whether there was an error. It's the key=value magic thing that these loggers use instead of string formatter verbs.

But valid point, I'll just use the if statement that's there and add an else to give clear wording instead. It's too important to the meaning to hide it in the metadata.